### PR TITLE
Added local microk8s cluster onboarding

### PR DESCRIPTION
**Pull request by shoegazerpt**
Closes microsoft/azure_arc#32

### CODE REVIEW

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+* Install MicroK8s following the [specific install guide](https://microk8s.io/) for your operating system.
+
+* Navigate to the folder that has the MicroK8s cluster definition.
+
+Windows:
+  ```bash
+  cd azure_arc\azure_arc_k8s_jumpstart\microk8s
+  ```
+
+Linux and Mac:
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:80]:
```suggestion
Linux and MacOS:
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  ```
+
+# Deployment
+
+* Install MicroK8s following the [specific install guide](https://microk8s.io/) for your operating system.
+
+* Navigate to the folder that has the MicroK8s cluster definition.
+
+Windows:
+  ```bash
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:76]:
```suggestion
  ```terminal
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+* Install the Azure Arc for Kubernetes CLI extensions ***connectedk8s*** and ***k8sconfiguration***:
+
+  ```bash
+  az extension add --name connectedk8s
+  az extension add --name k8sconfiguration
+  ```
+
+# Deployment
+
+* Install MicroK8s following the [specific install guide](https://microk8s.io/) for your operating system.
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:71]: Share a bit more here. See the kind example:
https://github.com/microsoft/azure_arc/blob/master/azure_arc_k8s_jumpstart/docs/local_kind.md.

Quick one-liners with Windows, Linux, MacOS

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+* Export MicroK8s config 
+
+MicroK8s will not update your .kube/config file, and accessing the cluster is done using the microk8s cli, eg: `microk8s kubectl get nodes`. To be able to use this config with the Azure Arc CLI, we need to export it into a file.
+
+Windows: 
+```
+microk8s config view > %HOMEPATH%\.kube\microk8s
+```
+Linux and Mac:
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:93]:
```suggestion
Linux and MacOS: 
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  ```bash
+  cd azure_arc\azure_arc_k8s_jumpstart\microk8s
+  ```
+
+Linux and Mac:
+  ```bash
+  cd azure_arc/azure_arc_k8s_jumpstart/microk8s
+  ```
+
+* Export MicroK8s config
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:85]:
```suggestion
* Export MicroK8s kubeconfig file path
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+```
+microk8s config view > ~/.kube/microk8s
+```
+
+# Connect the cluster to Azure Arc
+
+* Authenticate Azure CLI
+
+We'll start by authenticating our CLI with Azure, **replacing the values below** with the output from the command we issued to create service principal earlier (`az ad sp create-for-rbac`).
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:103]:
```suggestion
We'll start by authenticating our CLI with Azure, **replacing the values below** with the output from the command we issued to create service principal earlier (`az ad sp create-for-rbac`) in the prerequisite section of this guide.
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+```
+az login --service-principal -u appId -p password --tenant tenant
+```
+
+* Create a Resource Group
+
+```
+az group create -n Arc-MicroK8s-Demo -l EastUS
+```
+
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:114]: Add some screenshots (mask subscription ID)

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+* Connect the cluster to Azure Arc
+
+Windows:
+```
+az connectedk8s connect -n microk8s -g Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s  --tags 'Project=jumpstart_azure_arc_k8s'
+```
+
+Linux and Mac:
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:122]:
```suggestion
Linux and MacOS:
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+```
+az group create -n Arc-MicroK8s-Demo -l EastUS
+```
+
+* Connect the cluster to Azure Arc
+
+Windows:
+```
+az connectedk8s connect -n microk8s -g Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s  --tags 'Project=jumpstart_azure_arc_k8s'
+```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:120]:
```suggestion
az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s  --tags 'Project=jumpstart_azure_arc_k8s'
```
```

This is just to give a bit more context to the user around the name and the used resource group

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+Linux and Mac:
+```
+az connectedk8s connect -n microk8s -g Arc-MicroK8s-Demo  --kube-config ~/.kube/microk8s --kube-context microk8s  --tags 'Project=jumpstart_azure_arc_k8s'
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:124]:
```suggestion
az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo  --kube-config ~/.kube/microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    ```
+
+  ![](../img/local_microk8s/portal-delete-rg.png)
+
+* To stop the MicroK8s cluster locally, use the following command:
+    ```bash
+    microk8s stop
+    ```
+
+* To fully uninstall MicroK8s, check the [official docs](https://microk8s.io/docs) for your operating system.
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:151]: incl. the command in this case. You can refer to the official docs but let's include the `microk8s uninstall` command and a screenshot.

**Review by likamrat — CHANGES_REQUESTED** [review]
- Add screenshots. Also, change the img file names to numeric to follow the rest of the repo (01, 02, ..)
- Syntax updates
- Some extra context

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+    microk8s status --wait-ready
+    microk8s enable dns
+    ```
+
+  * Linux:
+    ```
+    sudo snap install microk8s --classic
+    microk8s status --wait-ready
+    microk8s enable dns
+    ```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:91]:
```suggestion
    ```

```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+
+  * Windows:
+  
+    Download the [MicroK8s installer for Windows](https://microk8s.io/docs/install-alternatives#heading--windows) and follow the instructions. 
+
+    ![](https://aws1.discourse-cdn.com/business6/uploads/kubernetes/original/2X/c/cc39a370d6a9f62bbcfa0b84ba8356da84a4c8c1.png)
+
+    Once installed, enable MicroK8s with:
+
+    ```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:81]:
```suggestion
    ```terminal
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+
+    Once installed, enable MicroK8s with:
+
+    ```terminal
+    microk8s status --wait-ready
+    microk8s enable dns
+    ```
+
+  * Linux:
+    ```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:87]:
```suggestion
    ```terminal
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+    microk8s status --wait-ready
+    microk8s enable dns
+    ```
+
+  * Linux:
+    ```terminal
+    sudo snap install microk8s --classic
+    microk8s status --wait-ready
+    microk8s enable dns
+    ```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:91]:
```suggestion
  * Linux:
    ```terminal
    sudo snap install microk8s --classic
    microk8s status --wait-ready
    microk8s enable dns
    
    ```
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+  ```
+
+  ![](../img/local_microk8s/02.png)
+
+* Connect the cluster to Azure Arc
+
+  Windows:
+  ```
+  az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
+  ```
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:158]:
```suggestion
  Windows:
  ```
  az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
  ```

```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+    ```terminal
+    cd azure_arc\azure_arc_k8s_jumpstart\microk8s
+    ```
+
+  Linux and MacOS:
+    ```bash
+    cd azure_arc/azure_arc_k8s_jumpstart/microk8s
+    ```
+
+* Export MicroK8s kubeconfig file path
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:119]:
```suggestion
* Export MicroK8s cluster kubeconfig file path
```

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+  ![](../img/local_microk8s/01.png)
+
+* Create a Resource Group
+
+  ```
+  az group create -n Arc-MicroK8s-Demo -l EastUS
+  ```
+
+  ![](../img/local_microk8s/02.png)
+
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:152]: Add a screenshot of an empty RG (mask subscription ID)

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 19]
```diff
@@ -0,0 +1,217 @@
...
+* Connect the cluster to Azure Arc
+
+  Windows:
+  ```
+  az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:157]:
```suggestion
  az connectedk8s connect --name Arc-MicroK8s-Demo --resource-group Arc-MicroK8s-Demo --kube-config %HOMEPATH%\.kube\microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
```

Update screenshot

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+  ```
+  Linux and MacOS:
+  ```
+  az connectedk8s connect --name microk8s --resource-group Arc-MicroK8s-Demo  --kube-config ~/.kube/microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/docs/local_microk8s.md:161]:
```suggestion
  az connectedk8s connect --name Arc-MicroK8s-Demo --resource-group Arc-MicroK8s-Demo  --kube-config ~/.kube/microk8s --kube-context microk8s --tags 'Project=jumpstart_azure_arc_k8s'
```

Update screenshot

**Review by likamrat — CHANGES_REQUESTED** [review]
- Update screenshots to meet the "Arc-MicroK8s-Demo" Arc cluster name
- img 06 mask subscription

File: azure_arc_k8s_jumpstart/docs/local_microk8s.md (hunk 18 above)
Comment (shoegazerpt) [reply azure_arc_k8s_jumpstart/docs/local_microk8s.md:152]: @likamrat done

**Review by likamrat — APPROVED** [review]
LGTM!

**Pull request merged**